src/game_set.ts

```diff
@@ -32,34 +36,244 @@ export default class GameSet extends Entity {
   }
 
   public updated(time_unit: number) {
-    const updatedRec = (game_set: GameSet, remaining: Immutable.List<number>): GameSet => {
-      if (remaining.size == 0)
-          return game_set;
+    const all_objects_except_ground = 
+      this.filter_contents(o => !o.is_ground)
+        .map(o => o.id)
+        .toList();
 
-      const first_key = remaining.first();
-      const first_object = game_set.contents.get(first_key) as GameElement;
+    return all_objects_except_ground.reduce(
+      (current_game_set, object_id) => current_game_set._updated_per_object(time_unit, current_game_set.contents.get(object_id)),
+      this as GameSet
+    ); 
+  }
+
+  private _updated_per_object(time_unit: number, object: PhysicalObject): GameSet {
+    const next_state = object.updated(time_unit) as PhysicalObject;
+    const next_state_resetted_translation = next_state.copy<PhysicalObject>({
+      position: object.position,
+      angle: object.angle
+    });
+
+    const delta_position = next_state.position.subtract(object.position);
+    const delta_angle = next_state.angle - object.angle;
+
+    class DeltaState {
```

Comment:
What does this do? And why does it have a `GameSet` reference? A better name would also be good.

Reply:
It's a data holder for when I reduce the list of contents, updating and colliding every one of them. Should those dummy data holders also be modeled as classes or a hacky json the way to go? (Feel I already know the answer .. )